packages/cdk-assets/lib/aws.ts

```diff
@@ -228,6 +228,8 @@ export class DefaultAwsClient implements IAws {
       config.region = options.region;
       if (options.assumeRoleArn) {
         config.credentials = fromTemporaryCredentials({
+          // dont forget the credentials chain.
+          masterCredentials: config.credentials,
```

Comment:
Similar to what we have in the CDK CLI:

https://github.com/aws/aws-cdk-cli/blob/1bef26344b3e311fbac7241ffa9a3e05b195d6de/packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts#L357

> This is also why the CDK CLI wasn't affected.

Reply:
Hmm - this might be problematic if there are multiple assets to upload because it creates a self-assume. Checking.

Reply:
Yeap, this was wrong.